Objects/tupleobject.c

```diff
@@ -119,7 +113,34 @@ PyTuple_New(Py_ssize_t size)
         if (op == NULL)
             return NULL;
     }
-    for (i=0; i < size; i++)
+    return (PyObject *) op;
+}
+
+_Py_NO_INLINE static PyObject *
+tuple_new_untracked(Py_ssize_t size)
```

Comment:
This function would deserve a comment :-) Especially, you must document that it must not be called with size=0 and that PyTuple_New(0) should be called instead.

Maybe explain that it's a micro-optimization that you should be used carefully ;-)

IMHO it must only be used if the code filling items cannot call arbitrary Python (like trigger a GC collection). Only if the code filling items is very simple, like _PyTuple_FromArray() which just copy strong references.